### PR TITLE
Add version checking

### PR DESCRIPTION
**Pull request by chrisvanrun**
This is a partial copy of the outdated PR of @HarmvZ who laid the foundations. However, it now also checks the latest supported version.

The original PR had some problems with double running the check and the new check needed access to the `base_url` anyway. So I decided put the checking in the `ClientBase` initialisation but keep it in a separate httpx Client.

### CODE REVIEW

File: gcapi/check_version.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    pass
+
+
+def check_version(base_url):
+    package_name = "gcapi"
+    try:
+        current_version = get_version(package_name)
+        with httpx.Client() as client:
+            response = client.get(f"https://pypi.org/pypi/{package_name}/json")
+            latest_version = response.json()["info"]["version"]
```
Comment (jmsmkn) [inline gcapi/check_version.py:18]: Now that we have the endpoint we should put this information into our endpoint and save both a call and a dependency on pypi being up.

File: gcapi/check_version.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+from packaging import version
+
+
+class UnsupportedVersionError(Exception):
+    pass
+
+
+def check_version(base_url):
+    package_name = "gcapi"
+    try:
```
Comment (jmsmkn) [inline gcapi/check_version.py:14]: I don't think the big try except clause is necessary if we do everything on our own servers now.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -8,7 +8,7 @@ repos:
       - id: mixed-line-ending
       - id: trailing-whitespace
   - repo: https://github.com/asottile/pyupgrade
-    rev: v3.18.0
+    rev: v3.19.0
```
Comment (jmsmkn) [inline .pre-commit-config.yaml:11]: Are these changes relevant to this PR? Would be nicer if these changes were separate.

**Review by jmsmkn — COMMENTED** [review]
Please fix the tests.

File: gcapi/check_version.py (hunk 1 above)
Comment (chrisvanrun) [reply gcapi/check_version.py:18]: Good point, how would you handle the versioning on the GC side though? Periodic pull from pypi or manual?

File: .pre-commit-config.yaml (hunk 3 above)
Comment (chrisvanrun) [reply .pre-commit-config.yaml]: Had a vague error, turns out my pre-commit was very outdated. We don't really need to update these now. Reverted.

File: gcapi/check_version.py (hunk 1 above)
Comment (jmsmkn) [reply gcapi/check_version.py:18]: It can be manual for now, taken from an environment variable, less dependencies on external systems the better. Don't think it's too much extra work given the frequency of releases. Then we could automate it later in terraform if needed.

**Comment by chrisvanrun** [comment]
(Forgot to push before reassigning.)

Small trick: autouse fixture to mock check_version during tests.

File: gcapi/check_version.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+def check_version(base_url):
+    package_name = "gcapi"
+
+    current_version = get_version(package_name)
+    with httpx.Client() as client:
+        response = client.get(f"{base_url}/gcapi/")
+
+        json = response.json()
+        latest_version = json["latest_version"]
+        lowest_supported_version = json["lowest_supported_version"]
```
Comment (jmsmkn) [inline gcapi/check_version.py:21]: Removes shadowing of standard library

```suggestion
    current_version = get_version(package_name)

    with httpx.Client() as client:
        response = client.get(f"{base_url}/gcapi/").json()

        api_data = response.json()

        latest_version = api_data["latest_version"]
        lowest_supported_version = api_data["lowest_supported_version"]
```

**Review by jmsmkn — APPROVED** [review]

File: gcapi/check_version.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    pass
+
+
+def check_version(base_url):
+    package_name = "gcapi"
+
+    current_version = get_version(package_name)
+
+    with httpx.Client() as client:
+        response = client.get(f"{base_url}/gcapi/").json()
```
Comment (jmsmkn) [inline gcapi/check_version.py:18]:
```suggestion
        response = client.get(f"{base_url}/gcapi/")
```

**Review by jmsmkn — APPROVED** [review]
